nixos/espeakup: init

ndarilek opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Added `services.espeakup` module which:
* Enables the `espeakup` systemd service
* Adds a `defaultVoice` configuration option for setting the espeakup default voice in the systemd unit. It defaults to "en-gb" as is typical with espeak.
* Adds "speakup_soft" to the boot kernel modules so speech synthesis is available when the kernel boots
* Clears the `ExecStartPre` of the unit as shipped--this just ran modprobe and is no longer necessary, and always errored out with an `exit(203)` when left in

I've tested that `services.espeakup.enable = true;` in my system configuration does in fact bring up espeakup on tty2. I'm running Pipewire and had to set `systemWide = true` so sound was available on the console, but it does speak and is responsive.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

FliegendeWurst commented on nixos/modules/services/accessibility/espeakup.nix:
```diff
@@ -0,0 +1,41 @@
+
+  config = mkIf cfg.enable {
+    boot.kernelModules = [ "speakup_soft" ];
+    systemd.packages = [ pkgs.espeakup ];
```
```suggestion
    systemd.packages = [ cfg.package ];
```

FliegendeWurst commented on nixos/modules/services/accessibility/espeakup.nix:
```diff
@@ -0,0 +1,41 @@
+      description = "Default voice for espeakup";
+    };
+  };
+
```
```suggestion
  meta.maintainers = with lib.maintainers; [ ndarilek ];
```

ndarilek committed:
maintainers: add ndarilek

ndarilek committed:
nixos/espeakup: init

Add a NixOS module for espeakup, a connector for the Speakup kernel
screen reader and the espeak-ng text-to-speech synthesizer.

The module:
- Enables the espeakup systemd service
- Loads the speakup_soft kernel module at boot
- Provides a defaultVoice option for setting the espeak voice
